src/lib/core/fiber.c

```diff
@@ -2339,3 +2350,18 @@ fiber_lua_state(struct fiber *f)
 {
 	return f->storage.lua.stack;
 }
+
+void
+fiber_shutdown(void)
+{
+	assert(cord()->shutdown_fiber == NULL);
+	struct fiber *fiber;
+	rlist_foreach_entry(fiber, &cord()->alive, link) {
+		if (!(fiber->flags & FIBER_IS_SYSTEM))
+			fiber_cancel(fiber);
+	}
+	cord()->shutdown_fiber = fiber();
+	while (cord()->client_fiber_count != 0)
+		fiber_yield();
+	cord()->shutdown_fiber = NULL;
```

Comment:
It may be more appropriate to use a condition variable here. Anyway, what happens, if a client fiber starts another client fiber after it's cancelled?

Reply:
Discussed f2f and decided to keep it like this (rationale: we don't want to bring `fiber_cond.h` dependence here).